src/main/java/mil/dds/anet/resources/EntityAvatarResource.java

```diff
@@ -0,0 +1,112 @@
+    return entityAvatarDao.getByRelatedObjectUuid(relatedObjectUuid).orElse(null);
+  }
+
+  /**
+   * Creates or updates an entity avatar
+   * 
+   * @param context the context
+   * @param entityAvatar the entity avatar
+   * @return the entity avatar
+   */
+  @GraphQLMutation(name = "createOrUpdateEntityAvatar")
+  public Integer createOrUpdateEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "entityAvatar") EntityAvatar entityAvatar) {
+    final int numRows;
+    final Person user = DaoUtils.getUserFromContext(context);
+
+    // Check if user is authorized to manipulate avatars for this relatedObjectUuid
+    assertPermission(user, entityAvatar.getRelatedObjectType(),
+        entityAvatar.getRelatedObjectUuid());
+
+    // Do we have an avatar already for this entity?
+    final Optional<EntityAvatar> dbEntityAvatar =
+        entityAvatarDao.getByRelatedObjectUuid(entityAvatar.getRelatedObjectUuid());
+
+    if (dbEntityAvatar.isEmpty()) {
+      numRows = entityAvatarDao.insert(entityAvatar);
+      AnetAuditLogger.log("Entity avatar created by {}", user);
+    } else {
+      numRows = entityAvatarDao.update(entityAvatar);
+      AnetAuditLogger.log("Entity avatar updated by {}", user);
+    }
+
+    // GraphQL mutations *have* to return something, so we return the number of updated rows
+    return numRows;
+  }
+
+  /**
+   * Deletes the entity avatar if any
+   * 
+   * @param context the context
+   * @param relatedObjectType the relatedObjectType
+   * @param relatedObjectUuid the relatedObjectUuid
+   * @return the number of rows deleted
+   */
+  @GraphQLMutation(name = "deleteEntityAvatar")
+  public Integer deleteEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "relatedObjectType") String relatedObjectType,
+      @GraphQLArgument(name = "relatedObjectUuid") String relatedObjectUuid) {
+    final Person user = DaoUtils.getUserFromContext(context);
+
+    // Check if user is authorized to manipulate avatars for this relatedObjectUuid
+    assertPermission(user, relatedObjectType, relatedObjectUuid);
+
+    int numRows = entityAvatarDao.delete(relatedObjectUuid);
+    if (numRows > 1) {
+      AnetAuditLogger.log("Entity avatar deleted by {}", user);
+    }
+
+    // GraphQL mutations *have* to return something, so we return the number of updated rows
+    return numRows;
+  }
+
+  public static void assertPermission(final Person user, final String relatedObjectType,
+      final String relatedObjectUuid) {
+    if (!hasPermission(user, relatedObjectType, relatedObjectUuid)) {
+      throw new WebApplicationException(AuthUtils.UNAUTH_MESSAGE, Response.Status.FORBIDDEN);
+    }
+  }
+
+  public static boolean hasPermission(final Person user, final String relatedObjectType,
+      final String relatedObjectUuid) {
+    if (relatedObjectType.equals("organizations")) {
```

Comment:
Prefer to use `OrganizationDao.TABLE_NAME` instead of the hard-coded string `"organizations"`.